client/rest-high-level/src/main/java/org/elasticsearch/client/SnapshotClient.java

```diff
@@ -67,4 +67,28 @@ public void getRepositoriesAsync(GetRepositoriesRequest getRepositoriesRequest,
         restHighLevelClient.performRequestAsyncAndParseEntity(getRepositoriesRequest, RequestConverters::getRepositories,
             GetRepositoriesResponse::fromXContent, listener, emptySet(), headers);
     }
+
+    /**
+     * Creates a snapshot repository.
+     * <p>
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html"> Snapshot and Restore
+     * API on elastic.co</a>
+     */
+
+    public PutRepositoryResponse putRepository(PutRepositoryRequest putRepositoryRequest, Header... headers) throws IOException {
+        return restHighLevelClient.performRequestAndParseEntity(putRepositoryRequest, RequestConverters::putRepository,
+            PutRepositoryResponse::fromXContent, emptySet(), headers);
+    }
+
+    /**
+     * Asynchronously creates a snapshot repository.
+     * <p>
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html"> Snapshot and Restore
+     * API on elastic.co</a>
+     */
+    public void putRepositoryAsync(PutRepositoryRequest putRepositoryRequest,
```

Comment:
in the spec we call this "snapshot.create_repository" . Could you rename the methods to createRepository please?